src/nodes.coffee

```diff
@@ -3551,6 +3553,33 @@ exports.Code = class Code extends Base
 
     seenSuper
 
+  propagateLhs: ->
+    for {name} in @params when name instanceof Arr or name instanceof Obj
+      name.propagateLhs yes
+
+  astType: -> 'FunctionExpression'
+
+  paramForAst: (param) ->
+    return param if param instanceof Expansion
+    {name, value, splat} = param
+    if splat
+      new Splat name, lhs: yes, postfix: splat.postfix
+      .withLocationDataFrom name
+    else if value?
+      new Assign name, value, null, param: yes
+      .withLocationDataFrom locationData: mergeLocationData name.locationData, value.locationData
+    else
+      name
+
+  astProperties: (o) ->
+    return
+      params: @paramForAst(param).ast(o) for param in @params
+      body: @body.ast o
+      generator: !!@isGenerator
+      async: !!@isAsync
+      bound: !!@bound
+      id: null
```

Comment:
Why `id: null`?

Reply:
`id` is used for named functions - since we never generated named functions, we always want `id: null` (which matches the Babel AST for both arrow functions `() => {}` and anonymous function expressions `function() {}`)

Reply:
That sounds fine, do you mind please adding a comment explaining this?

Reply:
Sure, added a comment